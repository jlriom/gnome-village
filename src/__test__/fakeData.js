export const category = {
    category: "category",
    count: 0
}

export const habitantsByHairColor = [
    {
        hair_color: 'Black',
        count: 256
    },
    {
        hair_color: 'Gray',
        count: 258
    },
    {
        hair_color: 'Green',
        count: 267
    },
    {
        hair_color: 'Pink',
        count: 285
    },
    {
        hair_color: 'Red',
        count: 273
    }
];

export const habitantsByProfession = [
    {
        profession: 'Baker',
        count: 214
    },
    {
        profession: 'Blacksmith',
        count: 180
    },
    {
        profession: 'Brewer',
        count: 200
    },
    {
        profession: 'Butcher',
        count: 210
    },
    {
        profession: 'Carpenter',
        count: 216
    },
    {
        profession: 'Cook',
        count: 190
    },
    {
        profession: 'Farmer',
        count: 205
    },
    {
        profession: 'Gemcutter',
        count: 186
    },
    {
        profession: 'Leatherworker',
        count: 185
    },
    {
        profession: 'Marble Carver',
        count: 190
    },
    {
        profession: 'Mason',
        count: 214
    },
    {
        profession: 'Mechanic',
        count: 182
    },
    {
        profession: 'Medic',
        count: 187
    },
    {
        profession: 'Metalworker',
        count: 192
    },
    {
        profession: 'Miner',
        count: 201
    },
    {
        profession: 'Potter',
        count: 202
    },
    {
        profession: 'Prospector',
        count: 202
    },
    {
        profession: 'Sculptor',
        count: 182
    },
    {
        profession: 'Smelter',
        count: 203
    },
    {
        profession: 'Stonecarver',
        count: 187
    },
    {
        profession: 'Tailor',
        count: 199
    },
    {
        profession: 'Tax inspector',
        count: 192
    },
    {
        profession: ' Tinker',
        count: 202
    },
    {
        profession: 'Woodcarver',
        count: 199
    }
];

export const auth = { name: "name", isGuest: false };

export const habitant = {
    id: 362,
    name: 'Cogwitz Bittergrill',
    thumbnail: 'http://www.publicdomainpictures.net/pictures/120000/nahled/white-hen.jpg',
    age: 127,
    weight: 40.074535,
    height: 93.66314,
    hair_color: 'Black',
    professions: [
        'Prospector',
        ' Tinker',
        'Baker',
        'Tailor',
        'Leatherworker',
        'Sculptor'
    ],
    friends: [
        'Whitwright Fussbuster',
        'Fizwood Bitterrocket',
        'Fizwood Felpower'
    ]
};

export const friends = [
    {
        id: 363,
        name: 'Cogwitzs friend ',
        thumbnail: 'http://www.publicdomainpictures.net/pictures/120000/nahled/white-hen.jpg',
        age: 127,
        weight: 40.074535,
        height: 93.66314,
        hair_color: 'Black',
        professions: [
            'Prospector',
            ' Tinker'
        ],
        friends: [
            'Whitwright Fussbuster'
        ]
    }
];
