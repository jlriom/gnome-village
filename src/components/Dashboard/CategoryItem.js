import React from "react";

const CategoryItem = ({ category, count }) => (
	<li className='list-group-item '>
		<span
			style={{
				display: "flex",
				justifyContent: "space-between"
			}}>
			<span>{category} </span>
			<span className='badge  badge-pill badge-secondary '>{count}</span>
		</span>
	</li>
);

export default CategoryItem;
