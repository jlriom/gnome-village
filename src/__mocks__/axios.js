import * as fakeData from "./fakeData";

export default {
	get: jest.fn(() =>
		Promise.resolve({
			data: fakeData.habitants
		})
	),
	post: jest.fn(() => Promise.resolve({ data: fakeData.habitant })),
	put: jest.fn(() => Promise.resolve({ data: fakeData.habitant })),
	delete: jest.fn(() => Promise.resolve({ data: fakeData.habitant }))
};
