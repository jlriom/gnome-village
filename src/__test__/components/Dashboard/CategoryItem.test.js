import React from "react";
import CategoryItem from "../../../components/Dashboard/CategoryItem";
import renderer from "react-test-renderer";
import { category } from "../../../__mocks__/fakeData";

describe("<CategoryItem />", () => {
	it("renders ok", () => {
		const tree = renderer.create(<CategoryItem {...category} />);
		expect(tree).toMatchSnapshot();
	});
});
