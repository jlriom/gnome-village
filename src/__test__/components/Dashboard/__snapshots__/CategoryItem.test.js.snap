// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CategoryItem /> renders ok 1`] = `
<li
  className="list-group-item "
>
  <span
    style={
      Object {
        "display": "flex",
        "justifyContent": "space-between",
      }
    }
  >
    <span>
      category
       
    </span>
    <span
      className="badge  badge-pill badge-secondary "
    >
      0
    </span>
  </span>
</li>
`;
